src/main/java/io/supertokens/webserver/api/oauth/OAuthTokenAPI.java

```diff
@@ -0,0 +1,110 @@
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.session.jwt.JWT.JWTException;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OAuthTokenAPI extends WebserverAPI {
+
+    public OAuthTokenAPI(Main main) {
+        super(main, RECIPE_ID.OAUTH.toString());
+    }
+
+    @Override
+    public String getPath() {
+        return "/recipe/oauth/token";
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        String iss = InputParser.parseStringOrThrowError(input, "iss", false); // input validation
+        JsonObject bodyFromSDK = InputParser.parseJsonObjectOrThrowError(input, "inputBody", false);
+
+        Boolean useStaticKeyInput = InputParser.parseBooleanOrThrowError(input, "useStaticSigningKey", true);
+
+        try {
+            OAuthProxyHelper.proxyFormPOST(
+                main, req, resp,
+                getAppIdentifier(req),
+                enforcePublicTenantAndGetPublicTenantStorage(req),
+                "/oauth2/token", // proxyPath
+                false, // proxyToAdmin
+                false, // camelToSnakeCaseConversion
+                () -> {
+                    Map<String, String> formFields = new HashMap<>();
+                    for (Map.Entry<String, JsonElement> entry : bodyFromSDK.entrySet()) {
+                        formFields.put(entry.getKey(), entry.getValue().getAsString());
+                    }
+            
+                    return formFields;
+                },
+                HashMap::new,
+                (statusCode, headers, rawBody, jsonBody) -> {
+                    if (jsonBody == null) {
+                        throw new IllegalStateException("unexpected response from hydra");
+                    }
+
+                    try {
+                        AppIdentifier appIdentifier = getAppIdentifier(req);
+                        Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+                        JsonObject accessTokenUpdate = InputParser.parseJsonObjectOrThrowError(bodyFromSDK, "access_token", true);
+                        JsonObject idTokenUpdate = InputParser.parseJsonObjectOrThrowError(bodyFromSDK, "id_token", true);
```

Comment:
I think this should be on the top level instead of inside `inputBody`, similar to `iss`. `inputBody` should be mapped to whatever we get from the frontend.

Reply:
Done